feature-libs/organization/account-summary/core/facade/account-summary.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { RoutingService, UserIdService } from '@spartacus/core';
+import {
+  AccountSummaryDetails,
+  AccountSummaryFacade,
+  AccountSummaryList,
+  DocumentQueryParams,
+} from '@spartacus/organization/account-summary/root';
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, map, shareReplay } from 'rxjs/operators';
+import { AccountSummaryConnector } from '../connectors';
+@Injectable({
+  providedIn: 'root',
+})
+export class AccountSummaryService implements AccountSummaryFacade {
+  userId: string;
+  orgUnit: string;
+
+  constructor(
+    private routingService: RoutingService,
+    private userIdService: UserIdService,
+    private accountSummaryConnector: AccountSummaryConnector
+  ) {
+    this.userIdService.takeUserId().subscribe((userId) => {
+      this.userId = userId;
+    });
+    this.routingService
+      .getRouterState()
+      .pipe(
+        map((routingData) => routingData.state.params),
+        distinctUntilChanged()
+      )
+      .subscribe((params) => (this.orgUnit = params.orgUnit));
+  }
+
+  getAccountSummary(): Observable<AccountSummaryDetails> {
```

Comment:
For us, we get `org unit id` from routing, but we don't know how customer get this id. So, I think it is better that `getAccountSummary()` can have the id as parameter. The caller pass the `org unit id` to it.

Reply:
not sure I agree, the orgUnit and current user is abstracted in the service because we get the account summary only based on the current logged in user and the created orgUnit for that user. It should not be possible to make calls with it using different orgUnits or logged in users.

Reply:
Unless we want to call account summary header details in other pages, but I doubt we would want that? Let me know what you think

Reply:
Ok added optional orgUnit parameter to make it easier for customer to be able to use header data.